lfs/transfer_queue.go

```diff
@@ -17,45 +18,52 @@ const (
 )
 
 type Transferable interface {
-	Check() (*api.ObjectResource, error)
-	Transfer(progress.CopyCallback) error
-	Object() *api.ObjectResource
 	Oid() string
 	Size() int64
 	Name() string
+	Path() string
+	Object() *api.ObjectResource
 	SetObject(*api.ObjectResource)
+	// Legacy API check - TODO remove this and only support batch
+	LegacyCheck() (*api.ObjectResource, error)
 }
 
-// TransferQueue provides a queue that will allow concurrent transfers.
+// TransferQueue organises the wider process of uploading and downloading,
+// including calling the API, passing the actual transfer request to transfer
+// adapters, and dealing with progress, errors and retries
 type TransferQueue struct {
```

Comment:
Interestingly it was _because_ I pulled out the actual upload/download functionality to an external type that it ended up holding extra instance vars. While we could further split the type by changing the way the API part works, that wasn't what I needed to do for this PR - the API calling remains basically the same, it's the upload/download that is externalised. I suggest we do what you suggest when moving to your new API interface later.